models/quality_measures/intermediate/nqf0059_diabetes_hemoglobin_a1c/quality_measures__int_nqf0059_numerator.sql

```diff
@@ -0,0 +1,225 @@
+   )
+}}
+
+{%- set performance_period_end -%}
+(
+
+    select 
+        performance_period_end
+    from {{ ref('quality_measures__int_nqf0059__performance_period') }}
+
+)
+{%- endset -%}
+
+with denominator as (
+
+    select
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+    from {{ ref('quality_measures__int_nqf0059_denominator') }}
+
+)
+
+, hba1c_test_code as (
+
+    select
+          code
+        , code_system
+        , concept_name
+    From {{ref('quality_measures__value_sets')}}
+    where concept_name in  (
+        'HbA1c Laboratory Test'
+    )
+)
+
+, labs as (
+    select
+        patient_id
+        , result
+        , result_date
+        , collection_date
+        , source_code_type
+        , source_code
+        , normalized_code_type
+        , normalized_code
+    from {{ ref('quality_measures__stg_core__lab_result')}}
+
+)
+
+, qualifying_labs as (
+    select
+      labs.patient_id
+    , labs.result
+    , coalesce(collection_date,result_date) as evidence_date
+    , hba1c_test_code.concept_name
+    from labs
+    inner join hba1c_test_code
+      on ( labs.normalized_code = hba1c_test_code.code
+       and labs.normalized_code_type = hba1c_test_code.code_system )
+      or ( labs.source_code = hba1c_test_code.code
+       and labs.source_code_type = hba1c_test_code.code_system )
+    where coalesce(collection_date,result_date) < {{ performance_period_end }}
+
+)
+
+, recent_readings as (
+    select
+          patient_id
+        , evidence_date
+        , result
+    from qualifying_labs
+    qualify row_number() over(partition by patient_id order by evidence_date desc) = 1
+
+)
+
+, qualifying_patients as (
+
+    select
+        denominator.*
+        , recent_readings.* exclude patient_id
+    from denominator
+    left join recent_readings
+        on denominator.patient_id = recent_readings.patient_id
+
+)
+
+, test_not_performed as (
+
+    select
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+        , evidence_date
+        , 1 as performance_flag
+        , 1 as numerator_flag
+    from qualifying_patients
+    where 
+        (evidence_date not between performance_period_begin and performance_period_end)
+        or evidence_date is null
+
+)
+
+, valid_patients as (
+
+    select
+        *
+    from qualifying_patients
+    where evidence_date between performance_period_begin and performance_period_end
+
+)
+
+, readings_exceeding_9 as (
+
+    select
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+        , evidence_date
+        , 1 as performance_flag
+        , 1 as numerator_flag
+    from valid_patients
+    where result > 9.0
```

Comment:
BigQuery is complaining about the data type of "result". Adding this type cast to all of the readings CTEs resolves it.
```
 where cast(result as {{ dbt.type_numeric() }}) > 9.0
```